05/DB.test.js

```diff
@@ -1 +1,144 @@
-import DB from './DB';
\ No newline at end of file
+import DB from './DB';
+
+describe('DB', () => {
+	describe('insert', () => {
+		it('should return 1 when add 1 item', () => {
+			expect.assertions(1);
+			const db = new DB();
+
+			return db.insert({ a: 1, b: 2 }).then(() => {
+				expect(db._rows.length).toBe(1);
+			});
+		});
+
+		it('should reject when ID is not a number', () => {
+			expect.assertions(1);
+			const db = new DB();
+
+			return db.insert({ id: 'x', a: 1, b: 2 }).catch((err) => {
+				expect(err).toBe('ID can be only number!');
+			});
+		});
+
+		it('should reject when ID is duplicated', () => {
+			expect.assertions(1);
+			const db = new DB();
+
+			return db
+				.insert({ id: 1, a: 1, b: 2 })
+				.then(() => db.insert({ id: 1, a: 3, b: 4 }))
+				.catch((err) => {
+					expect(err).toBe("ID can't be duplicated!");
+				});
+		});
+	});
+
+	describe('select', () => {
+		it('should reject when ID not found', () => {
+			expect.assertions(1);
+			const db = new DB();
+
+			return db
+				.insert({ id: 1, a: 1, b: 2 })
+				.then(() => db.select(2))
+				.catch((err) => {
+					expect(err).toBe('ID not found');
+				});
+		});
+
+		it('should resolve when item exists', () => {
+			expect.assertions(1);
+			const db = new DB();
+
+			return db
+				.insert({ id: 1, a: 1, b: 2 })
+				.then(() => db.select(1))
+				.then((resp) => {
+					expect(resp).toBe(resp);
+				});
```

Comment:
Tutaj trochę nie poprawne sprawdzenie.
Skoro jest zwrócone `resp` to wiadomo, że `resp === resp`, ale nie wiem czy to `{ id: 1, a: 1, b: 2 }`.

Należałoby by utworzyć obiekt np. `const data = { id: 1, a: 1, b: 2 }` i potem przyrównać `resp === data`